content/en/docs/project-user-guide/application-workloads/statefulsets.md

```diff
@@ -7,4 +7,299 @@ description: 'Kubernetes StatefulSets'
+
+- **Environment Variables**
+
+  Supports configuring environment variables for Pod in the form of key-value pairs.
+
+  - name：Set the name of the environment variable.
+  - value：Set the value of the variable reference.
+  - type：Set the type of environment variables, support customization, configuration items, keys, and variable/variable references.
+
+  For more information about the command, please visit [Pod variable](https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/?spm=a2c4g.11186623.2.20.16704b3e9qHXPb).
+
+- **Container Security Context**
+
+  A security context defines privilege and access control settings for a Pod or Container. [Pod Security Policies](https://kubernetes.io/docs/concepts/policy/pod-security-policy/).
+
+- **Sync Host Timezone**
+
+  The time zone of the container will be consistent with that of the host after synchronization.
+
+![statefulsets](/images/docs/project-user-guide/workloads/statefulsets_form_2_container_finish.png)
+
+#### Step3.2: Other Setting
+
+- **Update Strategy**
+
+  The `.spec.updateStrategy` field of StatefulSet allows you to configure and disable automatic scrolling update Pod containers, tags, resource requests or restrictions, and annotations.
+
+  - **RollingUpdate (Recommended)**
+
+    When  `.spec.template` should update, The Pods in the StatefulSet are automatically deleted and create a new Pod to replace. When updating, these Pods are performed in reverse order, sequentially deleting, creating, and waiting for the Pod to become Ready before proceeding to the next Pod update.
+
+  - **OnDelete**
+
+    When  `.spec.template` should update, The Pods in the StatefulSet are not automatically updated. The Pod must be manually deleted so that the controller can create a new Pod to react to its update.
+
+  For more information about the update strategy, please visit [Statefulset Update-Strategies](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#update-strategies).
+
+- **The number of Pods when updated**
+
+  The Partitions is mean which pods you want to update through the rolling update. It specifying by `.spec.updateStrategy.rollingUpdate.partition`. The default is 0. For more information about Partitions, please visit [Partitions](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#partitions).
+
+- **Pod Security Context**
+
+  A security context defines privilege and access control settings for a Pod or Container. For more information about Pod Security Policies, please visit [Pod Security Policies](https://kubernetes.io/docs/concepts/policy/pod-security-policy/).
+
+- **Deployment Mode**
+
+  Select different deployment modes to change the affinity and anti-affinity of the inter-pod. In k8s, the affinity is used `podAffinity` and the anti-affinity is used `podAntiAffinity`. In the Kubesphere both affinity and anti-affinity are set `preferredDuringSchedulingIgnoredDuringExecution`.
```

Comment:
->
You can select different deployment modes to switch between inter-pod affinity and inter-pod anti-affinity. In Kubernetes, inter-pod affinity is specified as field `podAffinity` of field `affinity` while inter-pod anti-affinity is specified as field `podAntiAffinity` of field `affinity`. In KubeSphere, both `podAffinity` and `podAntiAffinity` are set to `preferredDuringSchedulingIgnoredDuringExecution`. You can enable **Edit Mode** in the top right corner to see field details.